python/instrumentation/openinference-instrumentation-bedrock/src/openinference/instrumentation/bedrock/__init__.py

```diff
@@ -158,7 +164,75 @@ def instrumented_response(*args: Any, **kwargs: Any) -> Dict[str, Any]:
                 span.set_attributes(dict(get_attributes_from_context()))
                 return response  # type: ignore
 
-        return instrumented_response
+        @wraps(wrapped_client.converse)
+        def instrumented_response_converse(*args: Any, **kwargs: Any) -> Dict[str, Any]:
+            if context_api.get_value(_SUPPRESS_INSTRUMENTATION_KEY):
+                return wrapped_client._unwrapped_converse(*args, **kwargs)  # type: ignore
+
+            with tracer.start_as_current_span("bedrock.converse") as span:
+                span.set_attribute(
+                    SpanAttributes.OPENINFERENCE_SPAN_KIND,
+                    OpenInferenceSpanKindValues.LLM.value,
+                )
+
+                if model_id := kwargs.get("modelId"):
+                    _set_span_attribute(span, SpanAttributes.LLM_MODEL_NAME, model_id)
+
+                if request_messages := kwargs.get("messages"):
+                    # Extract the most recent request message in message history list
+                    current_request_message = request_messages[-1]
```

Comment:
why do we only get the last message here?

Reply:
err is this coming in a follow up pr?

Reply:
The converse API supports a back-and-forth conversation with LLMs. Therefore, "messages" in the request call is a list that contains all previous chat history. Since the current span is tracking the most recent prompt, I have it so that we get the last prompt in that list. However, this can be changed so that we include all previous prompt/responses from the chat history in the span

Reply:
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-runtime/client/converse.html#

Reply:
Yeah i think typically we get all the messages sent on every call since that span will be describing what happens when all of those messages are sent not just one of them. Even though it may be implied by other spans that this call was only adding this one message, it's still the case that this call contained all of the messages which affects teh output

Reply:
Resolved. Chat context is stored as llm.input_messages. Since there is no semantic convention available for system prompts, converse called with system prompts adds a message with the role "system" at the start of llm.input_messages.